common/utils.c

```diff
@@ -416,3 +416,94 @@ TDNFYesOrNo(
     return dwError;
 }
 
+uint32_t TDNFUriIsRemote(
+    const char* pszKeyUrl,
+    int *pnRemote
+)
+{
+    uint32_t dwError = 0;
+    const char *szRemotes[] = {"http://", "https://", "ftp://",
+                               "ftps://", NULL};
+    int i = 0;
+
+    *pnRemote = 0;
+    for(i = 0; szRemotes[i]; i++) {
+        if (strncasecmp(pszKeyUrl, szRemotes[i], strlen(szRemotes[i])) == 0) {
+            *pnRemote = 1;
+            break;
+        }
+    }
+    if (szRemotes[i] == NULL && strncasecmp(pszKeyUrl, "file://", 7) != 0) {
+       dwError = ERROR_TDNF_INVALID_PARAMETER;
+       BAIL_ON_TDNF_ERROR(dwError);
+    }
+cleanup:
+    return dwError;
+
+error:
+    goto cleanup;
+}
+
+uint32_t TDNFFileNameFromUri(
+    const char* pszKeyUrl,
+    char** ppszFile)
+{
+    uint32_t dwError = 0;
+    const char* pszPath = NULL;
+    char *pszFile = NULL;
+    size_t nOffset;
+    const char *szPrefixes[] = {"http://", "https://", "ftp://",
+                                "ftps://", "file://", NULL};
+    int i = 0;
+
+    for(i = 0; szPrefixes[i]; i++) {
+        if (strncasecmp(pszKeyUrl, szPrefixes[i], strlen(szPrefixes[i])) == 0) {
+            nOffset = strlen(szPrefixes[i]);
+            break;
+        }
+    }
+    if (szPrefixes[i] == NULL) {
+       dwError = ERROR_TDNF_INVALID_PARAMETER;
```

Comment:
same here : let me know your thoughts if we set some specific error here other than invalid parameters?